### PR TITLE
feat(payment): INT-6715 SquareV2: Add payment method component

### PR DESCRIPTION
**Pull request by mauricio-sg**
## What? [INT-6715](https://bigcommercecloud.atlassian.net/browse/INT-6715) and [INT-6756](https://bigcommercecloud.atlassian.net/browse/INT-6756)
Add a new payment method component for SquareV2

## Why?
To support [Square's Web Payments SDK](https://developer.squareup.com/docs/web-payments/overview)

## Testing / Proof
_Default styles:_
Screen Shot 2022-10-13 at 1 13 37 PM
_Custom styles:_
Screen Shot 2022-10-13 at 1 20 44 PM

### Depends on:
👉 https://github.com/bigcommerce/checkout-sdk-js/pull/1642 ~Therefore, circleci will fail due to property `squarev2` does not exist in type `PaymentInitializeOptions`~

@bigcommerce/apex-integrations @bigcommerce/checkout

### CODE REVIEW

**Comment by bc-lessgarmi** [comment]
💚 Tested!
**QE** details:
The task for Square V2 has been validated without issue.
The following was tested:

CP01_T6756_SquareV2_Look&Feel_ExperimentON
CP02_T6756_SquareV2_Look&Feel_ExperimentOFF
CP03_T6756_SquareV2_Look&Feel_LabelText
CP04_T6756_SquareV2_Look&Feel_FieldText
CP05_T6756_SquareV2_Look&Feel_FieldPlaceholderText
CP06_T6756_SquareV2_Look&Feel_FieldErrorText
CP07_T6756_SquareV2_Look&Feel_Background
CP08_T6756_SquareV2_Look&Feel_fieldInnerShadow
CP09_T6756_SquareV2_Look&Feel_FieldBorder
CP10_T6756_SquareV2_Look&Feel_CompleteOrder
CP11_T6756_SquareV2_Look&Feel_Sandbox
CP11_T6756_SquareV2_Look&Feel_CardLogo

Please see the attachment word below with the evidence:
https://drive.google.com/drive/folders/10-fMBfk7jS_CXZB_Qb34Sl-kkKOPr6mA?usp=sharing - Connect to preview

Please let me know any comment or doubt

**Review by ggarcia-bc — APPROVED** [review]
LGTM 👍🏼

File: packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx
Comment (animesh1987) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx:73]: Can we move this method outside the component body? So it's not recreated on every render.

File: packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx
Comment (animesh1987) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx:80]: How is squareV1 connected to Mollie?

File: packages/squarev2-integration/e2e/__har__/SquareV2-in-Payment-Step_656806659/recording.har
Comment (animesh1987) [inline packages/squarev2-integration/e2e/__har__/SquareV2-in-Payment-Step_656806659/recording.har:1418]: Can you please remove the `googleRecaptchaSitekey` from the response here.

File: packages/squarev2-integration/project.json
Comment (animesh1987) [inline packages/squarev2-integration/project.json:10]: Since we have a `js` mock file as well, can you please include that too.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.spec.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.spec.tsx:49]: Can we add this rules in `eslintrc` file for spec files, so we have a central place of noting overrides. And therefore clean them up once we upgrade jest.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.spec.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.spec.tsx:142]: Same as above.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:17]: Can you please explain the use of `dummy` form fields?

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:135]: This should always be declared at top of the component.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:141]: Same as above.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:177]: Can we move all the useEffect calls to top of the component? Also maybe move them into a custom hook?

File: packages/ui/src/utils/dom/getAppliedStyles.ts
Comment (animesh1987) [inline packages/ui/src/utils/dom/getAppliedStyles.ts:3]: Does this belong in `ui` package? If this is only used by square then it should be in that package.

File: packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx
Comment (mauricio-sg) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx:80]: It's `SquareOnPaymentMethodV1` not `SquareV1` I changed the name for clarification. We could say that are connected because they have the need to use the 'old' Payment Method component instead the new `V2` under certain circumstances. See #1086

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (mauricio-sg) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:177]: This one is not possible due to the `no-use-before-define` rule.

File: packages/ui/src/utils/dom/getAppliedStyles.ts
Comment (mauricio-sg) [reply packages/ui/src/utils/dom/getAppliedStyles.ts:3]: Do you mean, to move it to the Square package? I think it could be used by other packages in the future.

https://github.com/bigcommerce/checkout-js/blob/cdc374a082af0ae5a53c9de6255571eb3b937719/packages/core/src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx#L70

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:178]: Which variable does it complain regarding the no use before define?

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:26]: Sorry think I confused you :|, I was wondering if we can do better naming as compared to `dummy`. Are these just placeholder fields?

Comment (mauricio-sg) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx]: Can we name them `placeholderFields`?

File: packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx
Comment (animesh1987) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx:32]: We should have this check on line 1 for Mollie, and return `true`. That way we don't need to execute the remaining code for square.

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:48]: Maybe a line after this?

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [inline packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:1]: Do we need to worry about translations over here?

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx]: Yes, that sounds better. Thanks

File: packages/squarev2-integration/src/SquareV2PaymentMethod.tsx
Comment (animesh1987) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:1]: Discussed offline and square's SDK will take care of that.

Comment (mauricio-sg) [reply packages/squarev2-integration/src/SquareV2PaymentMethod.tsx:1]: AFAIK, Square's widget takes care of that. https://developer.squareup.com/reference/sdks/web/payments/objects/Payments#Payments.setLocale

**Review by bc-peng — APPROVED** [review]
LGTM, once we merge https://github.com/bigcommerce/checkout-sdk-js/pull/1737, this PR should be able to pass CI checks.
You might need to include SDK version bump in this PR or rebase.

**Pull request merged**
